dashboard/src/components/Header/ContextSelector.tsx

```diff
@@ -0,0 +1,134 @@
+import { IClustersState } from "../../reducers/cluster";
+import { CdsButton, CdsIcon } from "../Clarity/clarity";
+import useOutsideClick from "../js/hooks/useOutsideClick/useOutsideClick";
+
+import Column from "components/js/Column";
+import { definedNamespaces } from "shared/Namespace";
+import { app } from "shared/url";
+import Row from "../js/Row";
+import "./ContextSelector.css";
+
+ClarityIcons.addIcons(clusterIcon, fileGroupIcon, angleIcon);
+
+export interface IContextSelectorProps {
+  clusters: IClustersState;
+  defaultNamespace: string;
+  fetchNamespaces: () => void;
+  createNamespace: (ns: string) => Promise<boolean>;
+  getNamespace: (ns: string) => void;
+  setNamespace: (ns: string) => void;
+}
+
+function ContextSelector({
+  clusters,
+  defaultNamespace,
+  fetchNamespaces,
+  createNamespace,
+  getNamespace,
+  setNamespace,
+}: IContextSelectorProps) {
+  const dispatch = useDispatch();
+  const [open, setOpen] = useState(false);
+  const currentCluster = clusters.clusters[clusters.currentCluster];
+  const namespaceSelected = currentCluster.currentNamespace || defaultNamespace;
+  const [cluster, setCluster] = useState(clusters.currentCluster);
+  const [namespace, setNS] = useState(namespaceSelected);
+  // Control when users click outside
+  const ref = useRef(null);
+  useOutsideClick(setOpen, [ref], open);
+
+  useEffect(() => {
+    fetchNamespaces();
+    if (namespaceSelected !== definedNamespaces.all) {
+      getNamespace(namespaceSelected);
+    }
+  }, [fetchNamespaces, namespaceSelected, getNamespace]);
+
+  const toggleOpen = () => setOpen(!open);
+  const selectCluster = (event: React.ChangeEvent<HTMLSelectElement>) =>
+    setCluster(event.target.value);
+  const selectNamespace = (event: React.ChangeEvent<HTMLSelectElement>) =>
+    setNS(event.target.value);
+  const changeContext = () => {
+    setNamespace(namespace);
+    dispatch(push(app.apps.list(cluster, namespace)));
+    setOpen(false);
+  };
+
+  return (
+    <>
+      <div
+        className={`dropdown kubeapps-align-center kubeapps-dropdown ${open ? "open" : ""}`}
+        ref={ref}
+      >
+        <button className="kubeapps-nav-link" onClick={toggleOpen} aria-expanded={open}>
+          <Row>
+            <Column span={10}>
+              <span>Current Context</span>
+              <div>
+                <CdsIcon size="sm" shape="cluster" inverse={true} />
+                <span className="kubeapps-dropdown-text">{clusters.currentCluster}</span>
+                <CdsIcon size="sm" shape="file-group" inverse={true} />
+                <span className="kubeapps-dropdown-text">{namespaceSelected}</span>
+              </div>
+            </Column>
+            <Column span={2}>
+              <div
+                className={`kubeapps-align-center angle ${open ? "angle-opened" : "angle-closed"}`}
+              >
+                <CdsIcon shape="angle" inverse={true} />
+              </div>
+            </Column>
```

Comment:
Based on the design, I'm missing here some alignment and spacing between the items:

![Design for the selector where all the items are aligned and they have more spacing](https://user-images.githubusercontent.com/4056725/87010351-b6a25a00-c1c6-11ea-8bb3-7651a30ecfc4.png)

Also, any of the items is highlighted. I'd use a slightly bolder font for the selected items and a bit smaller font for the title.

Reply:
Updated:
![Screenshot from 2020-07-09 17-05-17](https://user-images.githubusercontent.com/4025665/87056788-66e38300-c206-11ea-9f86-761cdcb90b84.png)